apps/io-services-cms-webapp/src/lib/clients/jira-legacy-client.ts

```diff
@@ -125,6 +146,63 @@ export const jiraLegacyClient = (
     );
   };
 
+  const fetchApiWithRetry = (
+    url: string,
+    fetchApi: typeof fetch,
+    options: RequestInit,
+    defaultRetryAfter: number,
+    maxRetryCount: number
+  ): TE.TaskEither<Error, Response> => {
+    const attempt = (retryCount: number): TE.TaskEither<Error, Response> =>
+      pipe(
+        TE.tryCatch(() => fetchApi(url, options), E.toError),
+        TE.chain((response) =>
+          response.status === 429
+            ? pipe(
+                response.headers.get("Retry-After"),
+                O.fromNullable,
+                O.chain((delay) =>
+                  isNumber(Number(delay)) ? O.some(delay) : O.none
+                ),
+                O.fold(
+                  () => defaultRetryAfter,
+                  (retryAfter) => Number(retryAfter) * 1000
+                ),
+                (retryAfter) =>
+                  retryCount < maxRetryCount
+                    ? pipe(
+                        T.delay(retryAfter)(attempt(retryCount + 1)),
+                        TE.map(TE.right),
+                        TE.flatten
+                      )
+                    : TE.left(
+                        new Error(
+                          `Cannot Contact jira after retries, statusCode: ${
+                            response.status
+                          }, headers: ${extractHeaders(response)}`
+                        )
+                      )
+              )
+            : TE.right(response)
+        )
+      );
+
+    return attempt(0);
+  };
+
+  const extractHeaders = (response: Response) => {
+    try {
+      const headers: Record<string, string> = {};
+      response.headers.forEach((value, key) => {
+        // eslint-disable-next-line functional/immutable-data
+        headers[key] = value;
+      });
```

Comment:
Have you tried this??
```suggestion
      const headers = Object.fromEntries(response.headers.entries())
```